packages/eslint-plugin/scripts/codegen.ts

```diff
@@ -0,0 +1,208 @@
+[![BTC](https://img.shields.io/badge/BTC-f5f5f5?style=flat&colorA=000000&colorB=000000)](https://blockchain.com/btc/address/36fuguTPxGCNnYZSRdgdh6Ea94brCAjMbH)
+
+An ESLint plugin which provides lint rules for [@react-three/fiber](https://github.com/pmndrs/react-three-fiber).
+
+## Installation
+
+\`\`\`bash
+npm install @react-three/eslint-plugin --save-dev
+\`\`\`
+
+## Configuration
+
+Use the recommended [config](#recommended) to get reasonable defaults:
+
+\`\`\`json
+"extends": [
+  "plugin:@react-three/recommended"
+]
+\`\`\`
+
+If you do not use a config you will need to specify individual rules and add extra configuration.
+
+Add "@react-three" to the plugins section.
+
+\`\`\`json
+"plugins": [
+  "@react-three"
+]
+\`\`\`
+
+Enable the rules that you would like to use.
+
+\`\`\`json
+"rules": {
+  "@react-three/no-clone-in-frame-loop": "error"
+}
+\`\`\`
+
+## Rules
+
+✅ Enabled in the \`recommended\` [configuration](#recommended).<br>
+🔧 Automatically fixable by the \`--fix\` [CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).<br>
+💡 Manually fixable by [editor suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions).
+
+| Rule | Description | ✅ | 🔧 | 💡 | 
+| ---- | -- | -- | -- | -- |
+${rules
+  .map(
+    (rule) =>
+      `| ${link(rule.moduleName, rule.module.meta?.docs?.url)} | ${rule.module.meta?.docs?.description} | ${conditional(
+        '✅',
+        rule.module.meta?.docs?.recommended,
+      )} | ${conditional('🔧', rule.module.meta?.fixable)} | ${conditional('💡', rule.module.meta?.hasSuggestions)} |`,
+  )
+  .join('\n')}
+
+## Shareable configs
+
+<!-- This part of the readme is not currently codegen'd. If you add more configs make sure to update this. -->
+
+### Recommended
+
+This plugin exports a \`recommended\` configuration that enforces rules appropriate for everyone using React Three Fiber.
+
+\`\`\`json
+"extends": [
+  "plugin:@react-three/recommended"
+]
+\`\`\`
+
+### All
+
+This plugin also exports an \`all\` configuration that includes every available rule.
+
+\`\`\`json
+"extends": [
+  "plugin:@react-three/all"
+]
+\`\`\`
+  `
+
+  const filepath = join(srcDir, '../', 'README.md')
+  await writeFile(filepath, code)
+}
+
+async function generate() {
+  const rulePaths = await fs.readdir(rulesDir)
+  const recommended: FoundRule[] = []
+  const rules: FoundRule[] = []
+
+  for (const moduleName of rulePaths) {
+    if (ignore.includes(moduleName)) {
+      continue
+    }
+
+    const rule: Rule.RuleModule = (await import(join(rulesDir, moduleName))).default
+    const foundRule = { module: rule, moduleName: moduleName.replace(extname(moduleName), '') }
+    rules.push(foundRule)
+
+    if (rule.meta?.docs?.recommended) {
+      recommended.push(foundRule)
+    }
+  }
+
+  await generateRuleIndex(rules)
+  await generateConfig('all', rules)
```

Comment:
If we ever want to add another config it's just a matter of calling generate config again. Perhaps we could introduce a native config if that made sense?